src/hotspot/share/opto/gcm.cpp

```diff
@@ -1205,8 +1206,8 @@ Block* PhaseCFG::hoist_to_cheaper_block(Block* LCA, Block* early, Node* self) {
     }
 #endif
     cand_cnt++;
-    if (LCA_freq < least_freq              || // Better Frequency
-        (StressGCM && C->randomized_select(cand_cnt)) || // Should be randomly accepted in stress mode
+    if ((StressGCM && C->randomized_select(cand_cnt)) ||  // Should be randomly accepted in stress mode
+        LCA_freq < least_freq           ||    // Better Frequency
```

Comment:
Should we break once StressGCM chooses a block? Looks like:
```cpp
    cand_cnt++;
    bool use_random_block = StressGCM && C->randomized_select(cand_cnt);
    if (use_random_block ||  // Should be randomly accepted in stress mode
        LCA_freq < least_freq           ||    // Better Frequency
         (!StressGCM                    &&    // Otherwise, choose with latency
          !in_latency                   &&    // No block containing latency
          LCA_freq < least_freq * delta &&    // No worse frequency
          target >= end_lat             &&    // within latency range
          !self->is_iteratively_computed() )  // But don't hoist IV increments
             // because they may end up above other uses of their phi forcing
             // their result register to be different from their input.
       ) {
      least = LCA;            // Found cheaper block
      least_freq = LCA_freq;
      start_latency = start_lat;
      end_latency = end_lat;
      if (target <= start_lat)
        in_latency = true;
      if (use_random_block) {
        break;
      }
    }
  }
```